File: matlab/worldbank/worldbank_min.m
% Descargador Banco Mundial (JSON) — script mínimo
% ----------------------------------------------------------------------------
% Objetivo
%   Construir la URL JSON del Banco Mundial, descargar los datos, convertirlos
%   a tabla de MATLAB y exportar a CSV.
% Parámetros editables
%   - iso3: país(es) ISO-3; indicator: indicador(es)
%   - date: rango temporal 'YYYY:YYYY'
% Salida
%   - Archivo CSV con los registros
% Pasos
%   1) Construir URL con format=json&per_page=...
%   2) webread + jsondecode
%   3) struct2table y exportar a CSV

% Parámetros básicos
iso3 = 'ESP';
indicator = 'NY.GDP.MKTP.KD.ZG';
date = '2015:2020';

% Construir la URL JSON del Banco Mundial
base_url = 'https://api.worldbank.org/v2';
path = ['/country/' iso3 '/indicator/' indicator];
url = [base_url path '?format=json&per_page=20000&date=' date];

% Leer la URL directamente a tabla
txt = webread(url, weboptions('Timeout', 60, 'ContentType', 'text'));
obj = jsondecode(txt);
S = obj{2};

% Convertir a tabla
T = struct2table(S);

% Exportar la tabla a CSV
writetable(T, 'worldbank_min.csv', 'FileType', 'text');

File: python/worldbank/worldbank_min.py
# Descargador mínimo Banco Mundial (JSON -> CSV)
# ----------------------------------------------------------------------------
# Objetivo
#   Consulta indicadores del Banco Mundial y guarda un CSV
# ----------------------------------------------------------------------------
import os
import pandas as pd
import requests


def main() -> None:
    # Cambiar directorio a la carpeta del script para rutas consistentes
    os.chdir(os.path.dirname(os.path.abspath(__file__)))

    # Construcción de URL y petición
    base_url = "https://api.worldbank.org/v2"
    path = "/country/ESP/indicator/NY.GDP.MKTP.KD.ZG"
    url = f"{base_url}{path}"
    params = {"format": "json", "per_page": 20000, "date": "2000:2023"}
    resp = requests.get(url, params=params, headers={"Accept": "application/json"}, timeout=120)
    resp.raise_for_status()
    obj = resp.json()
    if not isinstance(obj, list) or len(obj) < 2 or obj[1] is None:
        raise RuntimeError("Unexpected World Bank response structure; no data array present.")
    pd.DataFrame(obj[1]).to_csv("worldbank_min.csv", index=False)


if __name__ == "__main__":
    main()


